chain-extensions/xvm/src/lib.rs

```diff
@@ -87,7 +93,30 @@ where
                 // Similar to EVM behavior, the `source` should be (limited to) the
                 // contract address. Otherwise contracts would be able to do arbitrary
                 // things on behalf of the caller via XVM.
-                let source = env.ext().address();
+                let source = env.ext().address().clone();
+
+                // Claim the default evm address if needed.
+                let mut actual_weight = Weight::zero();
+                if value > 0 {
+                    // `UA::to_h160` 1 DB read.
+                    actual_weight.saturating_accrue(T::DbWeight::get().reads(1));
+
+                    if UA::to_h160(&source).is_none() {
+                        let weight_of_claim = <T as pallet_unified_accounts::Config>::WeightInfo::claim_default_evm_address();
+                        actual_weight.saturating_accrue(weight_of_claim);
```

Comment:
I've difficult time understanding, what if `actual_weight ` after adding UA weight is higher than `weight_limit`?
As I understand, we don't have any check here to make sure such situation doesn't happen.

Reply:
I'm aware of this, but didn't want to make the weight handling overcomplicated. It only happens in strict conditions and the weight amount is small. To make it 100% accurate, the weight limit needs to be adjusted as well, and this procedure would repeat everywhere.